core/src/main/java/org/pentaho/platform/engine/services/connection/datasource/dbcp/PooledDatasourceHelper.java

```diff
@@ -322,6 +313,22 @@ public static PoolingDataSource setupPooledDataSource( IDatabaseConnection datab
     }
   }
 
+  protected static ConnectionFactory getConnectionFactory( IDatabaseConnection databaseConnection, String url ) {
+    ConnectionFactory factory = null;
+    if ( url.startsWith( "jdbc:mysql:" ) || ( url.startsWith( "jdbc:mariadb:" ) ) ) {
```

Comment:
We cannot do this - this is a really bad practice. Please don't write code that resembles this either. We need symmetric encoding/decoding that's done for all databases, not targeted at MySQL/MariaDB.